Design.md

```diff
@@ -0,0 +1,55 @@
+# Design doc for job-runner
+
+Job-runner provides a gRCP API to execute arbitrary commands on a Linux host.
+
+## Runner
+
+The jobs themselves are normal child processes of the runner. The runner configures CGroups and Linux namespaces for children to limit resource usage and isolate the them from the other processes. It also handles output recording and forwarding. All data is stored in-memory. In case of the service process termination, all jobs and their status are lost.
+
+## Communication and auth
+
+Clients communicate via gRPC with a simple protocol: One service with four endpoints. The communication is secured [RusTLS](https://github.com/ctz/rustls) which by design supports only modern, secure cipher suites. [Tonic](https://github.com/hyperium/tonic) is used to serve gRPC and almost automatically handles protocol buffers, encryption and related concerns.
```

Comment:
which TLS version and ciphers will be used?

Reply:
From the RusTLS link:
* TLS version 1.2 and TLS1.3
* Authentication with ECDSA, Ed25519 or RSA

I see no reason to specify the exact cipher as a design choice, but for instance RSA with 4096 key length should be reasonable.